girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -270,3 +285,105 @@ def cancelUpload(self, upload):
         """
         if os.path.exists(upload['tempFile']):
             os.unlink(upload['tempFile'])
+
+    def importFile(self, item, path, user, name=None, mimeType=None, size=None):
+        """
+        Import a single file from the filesystem into the assetstore.
+
+        :param item: The parent item for the file.
+        :type item: dict
+        :param path: The path on the local filesystem.
+        :type path: str
+        :param user: The user to list as the creator of the file.
+        :type user: dict
+        :param name: Name for the file. Defaults to the basename of ``path``.
+        :type name: str
+        :param mimeType: MIME type of the file if known.
+        :type mimeType: str
+        :param size: Size of the file in bytes. Will stat the file if not
+            passed.
+        :type size: int
+        :returns: The file document that was created.
+        """
+        if size is None:
+            size = os.path.getsize(path)
+
+        name = name or os.path.basename(path)
+
+        file = self.model('file').createFile(
+            name=name, creator=user, item=item, reuseExisting=True,
+            assetstore=self.assetstore, mimeType=mimeType, size=size)
+        file['fullPath'] = os.path.abspath(os.path.expanduser(path))
```

Comment:
Looking at other assetstore, path are stored differently in the `file` model. Would it make sense to re-use existing convention ? `path` or `relpath`.

* imported file:

file['fullPath']

* [s3_assetstore_adapter.py#L280](https://github.com/girder/girder/blob/7760491d004510efee39e2d9f8b8174d051853fa/girder/utility/s3_assetstore_adapter.py#L280):

```
file['relpath']
```

* [filesystem_assetstore_adapter.py#L209](https://github.com/girder/girder/blob/7760491d004510efee39e2d9f8b8174d051853fa/girder/utility/filesystem_assetstore_adapter.py#L209):

```
file['path']
````